### PR TITLE
Remove support for Python 3.9, add support for Python 3.14

### LINKED ISSUE #189: Heads up: Test `testrunner-edge-cases.rst` fails with Python 3.14.0a4
### What I did:

I tried to build zope.testrunner with Python 3.14.0a4 in Fedora Linux 43.
The test output is different than what's expected.

### What I expect to happen:
Test pass

### What actually happened:

```
+ /usr/bin/python3 -m tox --current-env -q --recreate -e py314
Running tests at level 1
Running zope.testrunner.layer.UnitTests tests:
  Set up zope.testrunner.layer.UnitTests in 0.000 seconds.
  Running:
.............

Failure in test /builddir/build/BUILD/python-zope-testrunner-6.6-build/BUILDROOT/usr/lib/python3.14/site-packages/zope/testrunner/tests/testrunner-edge-cases.rst
Failed doctest test for testrunner-edge-cases.rst
  File "/builddir/build/BUILD/python-zope-testrunner-6.6-build/BUILDROOT/usr/lib/python3.14/site-packages/zope/testrunner/tests/testrunner-edge-cases.rst", line 0

----------------------------------------------------------------------
File "/builddir/build/BUILD/python-zope-testrunner-6.6-build/BUILDROOT/usr/lib/python3.14/site-packages/zope/testrunner/tests/testrunner-edge-cases.rst", line 91, in testrunner-edge-cases.rst
Failed example:
    try: testrunner.run_internal(defaults)
    finally: sys.stdin = real_stdin
    # doctest: +ELLIPSIS
Differences (ndiff with -expected +actual):
    - Running zope.testrunner.layer.UnitTests tests:...
    ?                                               ---
    + Running zope.testrunner.layer.UnitTests tests:
    +   Set up zope.testrunner.layer.UnitTests in N.NNN seconds.
    - > testrunner-ex/sample3/sampletests_d.py(NNN)f()
    - -> ...
      (Pdb) p x
    - 1
      (Pdb) c
        Ran 1 tests with 0 failures, 0 errors and 0 skipped in N.NNN seconds.
    - ...
    + Tearing down left over layers:
    +   Tear down zope.testrunner.layer.UnitTests in N.NNN seconds.
      False
```
### What version of Python and Zope/Addons I am using:

Fedora Linux 43, Python 3.14.0a4, zope.testrunner 6.6

### LINKED ISSUE #203: `testrunner-colors.rst` fails with Python 3.14.2
### What I did:

I ran tests for `zope.testrunner 8.1` with Python 3.14.2 and the `testrunner-colors.rst` test failed.
The `testrunner-edge-cases.rst` test fails too, but that's well known problem - #189.

Steps to reproduce:
```
wget https://files.pythonhosted.org/packages/source/z/zope_testrunner/zope_testrunner-8.1.tar.gz
gtar xf zope_testrunner-8.1.tar.gz
cd zope_testrunner-8.1
tox -e py314 -- --test testrunner-colors
```

### What I expect to happen:

The test pass.

### What actually happened:

```
py314: commands[0]> zope-testrunner --test-path=src --test testrunner-colors
Running zope.testrunner.layer.UnitTests tests:
  Set up zope.testrunner.layer.UnitTests in 0.000 seconds.


Failure in test /tmp/test/zope_testrunner-8.1/src/zope/testrunner/tests/testrunner-colors.rst
Failed doctest test for testrunner-colors.rst
  File "/tmp/test/zope_testrunner-8.1/src/zope/testrunner/tests/testrunner-colors.rst", line 0

----------------------------------------------------------------------
File "/tmp/test/zope_testrunner-8.1/src/zope/testrunner/tests/testrunner-colors.rst", line 76, in testrunner-colors.rst
Failed example:
    testrunner.run_internal(defaults)
Differences (ndiff with -expected +actual):
      {normal}Running zope.testrunner.layer.UnitTests tests:{normal}
        Set up zope.testrunner.layer.UnitTests in {green}N.NNN{normal} seconds.
    - <BLANKLINE>
    - <BLANKLINE>
      {boldred}Failure in test eek (sample2.sampletests_e){normal}
      Failed doctest test for sample2.sampletests_e.eek
       testrunner-ex/sample2/sampletests_e.py", Line NNN, in eek
    + ----------------------------------------------------------------------
    - <BLANKLINE>
    - ...
      {normal}File testrunner-ex/sample2/sampletests_e.py{normal}", Line {boldred}NNN{normal}, in {boldcyan}sample2.sampletests_e.eek{normal}
      Failed example:
      {cyan}    f(){normal}
      Exception raised:
      {red}    Traceback (most recent call last):{normal}
    - {red}      File ".../doctest.py", Line NNN, in __run{normal}
    - {red}        ...{normal}
      {red}      File "<doctest sample2.sampletests_e.eek[0]>", Line NNN, in ?{normal}
      {red}        f(){normal}
      {red} testrunner-ex/sample2/sampletests_e.py", Line NNN, in f{normal}
      {red}        g(){normal}
      {red} testrunner-ex/sample2/sampletests_e.py", Line NNN, in g{normal}
      {red}        x = y + 1  # noqa: F821{normal}
      {red}       - __traceback_info__: I don't know what Y should be.{normal}
      {red}    NameError: name 'y' is not defined{normal}
    - <BLANKLINE>
    - <BLANKLINE>
    - <BLANKLINE>
    - {boldred}Error in test test3 (sample2.sampletests_e.Test...){normal}
    ?                                                          ^^
    + {boldred}Error in test test3 (sample2.sampletests_e.Test.test3){normal}
    ?                                                          ^^^^^
      Traceback (most recent call last):
      {normal} testrunner-ex/sample2/sampletests_e.py{normal}", Line {boldred}NNN{normal}, in {boldcyan}test3{normal}
      {cyan}    f(){normal}
      {normal} testrunner-ex/sample2/sampletests_e.py{normal}", Line {boldred}NNN{normal}, in {boldcyan}f{normal}
      {cyan}    g(){normal}
      {normal} testrunner-ex/sample2/sampletests_e.py{normal}", Line {boldred}NNN{normal}, in {boldcyan}g{normal}
      {cyan}    x = y + 1  # noqa: F821{normal}
      {red}   - __traceback_info__: I don't know what Y should be.{normal}
      {red}NameError: name 'y' is not defined{normal}
    - <BLANKLINE>
    - <BLANKLINE>
    - <BLANKLINE>
      {boldred}Failure testrunner-ex/sample2/e.rst{normal}
      Failed doctest test for e.rst
       testrunner-ex/sample2/e.rst", line 0  
    + ----------------------------------------------------------------------
    - <BLANKLINE>
    - ...
      {normal}File testrunner-ex/sample2/e.rst{normal}", Line {boldred}NNN{normal}, in {boldcyan}e.rst{normal}
      Failed example:
      {cyan}    f(){normal}
      Exception raised:
      {red}    Traceback (most recent call last):{normal}
    - {red}      File ".../doctest.py", Line NNN, in __run{normal}
    - {red}        ...{normal}
      {red}      File "<doctest e.rst[1]>", Line NNN, in ?{normal}
      {red}        f(){normal}
      {red}      File "<doctest e.rst[0]>", Line NNN, in f{normal}
      {red}        return x{normal}
      {red}    NameError: name 'x' is not defined{normal}
    - <BLANKLINE>
    - <BLANKLINE>
    - <BLANKLINE>
    - {boldred}Failure in test test (sample2.sampletests_f.Test...){normal}
    ?                                                           ^^
    + {boldred}Failure in test test (sample2.sampletests_f.Test.test){normal}
    ?                                                           ^^^^
      Traceback (most recent call last):
      {normal} testrunner-ex/sample2/sampletests_f.py{normal}", Line {boldred}NNN{normal}, in {boldcyan}test{normal}
      {cyan}    self.assertEqual(1, 0){normal}
      {red}AssertionError: 1 != 0{normal}
    - <BLANKLINE>
      {normal}  Ran {green}164{normal} tests with {boldred}3{normal} failures, {boldred}1{normal} errors, {green}0{normal} skipped in {green}N.NNN{normal} seconds.{normal}
    - ...
    + {normal}Running samplelayers.Layer1 tests:{normal}
    +   Tear down zope.testrunner.layer.UnitTests in {green}N.NNN{normal} seconds.
    +   Set up samplelayers.Layer1 in {green}N.NNN{normal} seconds.
    + {normal}  Ran {green}9{normal} tests with {green}0{normal} failures, {green}0{normal} errors, {green}0{normal} skipped in {green}N.NNN{normal} seconds.{normal}
    + {normal}Running samplelayers.Layer11 tests:{normal}
    +   Set up samplelayers.Layer11 in {green}N.NNN{normal} seconds.
    + {normal}  Ran {green}26{normal} tests with {green}0{normal} failures, {green}0{normal} errors, {green}0{normal} skipped in {green}N.NNN{normal} seconds.{normal}
    + {normal}Running samplelayers.Layer111 tests:{normal}
    +   Set up samplelayers.Layerx in {green}N.NNN{normal} seconds.
    +   Set up samplelayers.Layer111 in {green}N.NNN{normal} seconds.
    + {normal}  Ran {green}26{normal} tests with {green}0{normal} failures, {green}0{normal} errors, {green}0{normal} skipped in {green}N.NNN{normal} seconds.{normal}
    + {normal}Running samplelayers.Layer112 tests:{normal}
    +   Tear down samplelayers.Layer111 in {green}N.NNN{normal} seconds.
    +   Set up samplelayers.Layer112 in {green}N.NNN{normal} seconds.
    + {normal}  Ran {green}26{normal} tests with {green}0{normal} failures, {green}0{normal} errors, {green}0{normal} skipped in {green}N.NNN{normal} seconds.{normal}
    + {normal}Running samplelayers.Layer12 tests:{normal}
    +   Tear down samplelayers.Layer112 in {green}N.NNN{normal} seconds.
    +   Tear down samplelayers.Layerx in {green}N.NNN{normal} seconds.
    +   Tear down samplelayers.Layer11 in {green}N.NNN{normal} seconds.
    +   Set up samplelayers.Layer12 in {green}N.NNN{normal} seconds.
    + {normal}  Ran {green}26{normal} tests with {green}0{normal} failures, {green}0{normal} errors, {green}0{normal} skipped in {green}N.NNN{normal} seconds.{normal}
    + {normal}Running samplelayers.Layer121 tests:{normal}
    +   Set up samplelayers.Layer121 in {green}N.NNN{normal} seconds.
    + {normal}  Ran {green}26{normal} tests with {green}0{normal} failures, {green}0{normal} errors, {green}0{normal} skipped in {green}N.NNN{normal} seconds.{normal}
    + {normal}Running samplelayers.Layer122 tests:{normal}
    +   Tear down samplelayers.Layer121 in {green}N.NNN{normal} seconds.
    +   Set up samplelayers.Layer122 in {green}N.NNN{normal} seconds.
    + {normal}  Ran {green}26{normal} tests with {green}0{normal} failures, {green}0{normal} errors, {green}0{normal} skipped in {green}N.NNN{normal} seconds.{normal}
    + {normal}Tearing down left over layers:{normal}
    +   Tear down samplelayers.Layer122 in {green}N.NNN{normal} seconds.
    +   Tear down samplelayers.Layer12 in {green}N.NNN{normal} seconds.
    +   Tear down samplelayers.Layer1 in {green}N.NNN{normal} seconds.
      {normal}Total: {green}329{normal} tests, {boldred}3{normal} failures, {boldred}1{normal} errors, {green}0{normal} skipped in {green}N.NNN{normal} seconds.{normal}
      True
  Ran 1 tests with 1 failures, 0 errors and 0 skipped in 0.718 seconds.
Tearing down left over layers:
  Tear down zope.testrunner.layer.UnitTests in 0.000 seconds.
py314: exit 1 (1.12 seconds) /tmp/test/zope_testrunner-8.1> zope-testrunner --test-path=src --test testrunner-colors pid=366214
```

### What version of Python and Zope/Addons I am using:

OS: OpenIndiana
Python: 3.14.2
zope.testrunner: 8.1

### PR DESCRIPTION
**Pull request by dataflake**
This also shows test issues, some of them mentioned in https://github.com/zopefoundation/zope.testrunner/issues/189

Fixes #203.
Fixes #189.

### CODE REVIEW

**Comment by icemac** [comment]
@dataflake As you created this PR you are officially not able to review it. As I did most of the fixing work, could you please look into the PR and approve or reject it in text?

**Comment by dataflake** [comment]
Looks good to me. Do you need me to merge?

**Comment by icemac** [comment]
@dataflake I can merge and release myself, just wanted your okay for my code changes.

**Comment by dataflake** [comment]
Yes, everything is fine, go ahead

File: CHANGES.rst [hunk 1]
```diff
@@ -5,7 +5,9 @@
 8.2 (unreleased)
 ================
 
-- Nothing changed yet.
+- Remove support for Python 3.9.
+
+- Add support for Python 3.1.4.
```
Comment (icemac) [inline CHANGES.rst]:
```suggestion
- Add support for Python 3.14.
```

**Review by icemac — APPROVED** [review]
